cclib/parser/serenityparser.py

```diff
@@ -194,3 +206,82 @@ def extract(self, inputfile, line):
                 line = next(inputfile)
                 i += 1
             self.append_attribute("mpenergies", [line.split()[2]])
+
+        # TODO add CC2 etc
+        if line.strip().startswith("TDDFT Summary"):
+            self.metadata["excited_states_method"] = "TD-DFT"
+
+        # excitation energies and singly-excited configuration data
+        if line.strip().startswith("Dominant Contributions"):
+            self.skip_line(inputfile, ["Dominant"])
+            self.skip_line(inputfile, ["dashes"])
+            self.skip_line(inputfile, ["state"])
+            self.skip_line(inputfile, ["(a.u.)"])
+            line = next(inputfile)
+
+            exc_iterator = 1
+            transition_data = []
+            while not line.strip().startswith("--"):
+                if line.strip().startswith(str(exc_iterator)):  # new exc
+                    if exc_iterator > 1:
+                        self.append_attribute("etsecs", transition_data)
+                        transition_data = []
+                    self.append_attribute("etenergies", line.split()[1])
+                    i = int(line.split()[6])
```

Comment:
Store the results from `line.split()` in a variable that you reuse (this happens in a few places in the PR).